routers/routes/routes.go

```diff
@@ -277,6 +278,60 @@ func RegisterRoutes(m *macaron.Macaron) {
 		ctx.Data["UnitPullsGlobalDisabled"] = models.UnitTypePullRequests.UnitGlobalDisabled()
 	})
 
+	// get theme now
+	m.Use(func(ctx *context.Context) {
+		themeNow := ""
+		// Change theme handle
+		chooseTheme := ctx.Query("theme")
+
+		if len(chooseTheme) != 0 {
+			var exists bool
+
+			for _, v := range setting.UI.Themes {
+				if strings.EqualFold(strings.ToLower(v), strings.ToLower(chooseTheme)) {
+					exists = true
+					break
+				}
+			}
+
+			if exists {
+				themeNow = chooseTheme
+				ctx.SetCookie("themeNow", themeNow, nil, setting.AppSubURL, setting.SessionConfig.Domain, setting.SessionConfig.Secure, true)
+			} else {
+				ctx.Flash.Error(ctx.Tr("settings.theme_update_error"))
+			}
+		}
+
+		if themeNow == "" {
+			chooseTheme = ctx.GetCookie("themeNow")
+			var exists bool
+
+			for _, v := range setting.UI.Themes {
+				if strings.EqualFold(strings.ToLower(v), strings.ToLower(chooseTheme)) {
+					exists = true
+					break
+				}
+			}
+
+			if exists {
+				themeNow = chooseTheme
+			} else {
+				ctx.Flash.Error(ctx.Tr("settings.theme_update_error"))
```

Comment:
What about the error message itself?

Reply:
https://github.com/go-gitea/gitea/blob/209b17c4e25fe72d2fdf46f412fc388bc274a516/options/locale/locale_en-US.ini#L473